### PR TITLE
Http framework free

### PR DESCRIPTION
**Pull request by digitalsadhu**
This PR makes the proxy http framework free making it possible to integrate it in multiple http frameworks such as Hapi, Fastify etc.

Previously Podium has been very tied to Express.js where each module in the request chain has been a middleware where properties from each module has been written to `res.locals.podium`. Internally multiple of Express.js only methods have also been used making Podium very tied to Express.js.

This implementation switches from using a middleware chain passing request and response objects around to a Promise chain where a HttpIncoming object we have controll over is passed on. In other words; each module in Podium does now have a `.process()` method which can be chained together making it possible for any other http framwork to be integrated.

All usage of Express.js only methods has also been replaced with generic methods which should work within any http framework.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,17 @@
 # @podium/proxy
 
-Transparent http proxy. Dynamically mounts proxy targets on an existing HTTP server instance.
+Transparent http proxy. Dynamically mounts proxy targets on an existing HTTP
```
Comment (digitalsadhu) [inline README.md]: HTTP

File: README.md [hunk 2]
```diff
@@ -14,14 +20,12 @@ $ npm install @podium/proxy
 
 ## Simple usage
 
-Attach a proxy target to an existing Express server.
+Attach a proxy target to an http server.
```
Comment (digitalsadhu) [inline README.md]: HTTP

File: README.md [hunk 3]
```diff
@@ -97,36 +110,76 @@ proxy.register({
...
-A Podium manifest where the `proxy` property is given. The `proxy` property is an object where the `key` identifies the target and the `property` is a URI to the target.
+A Podium manifest where the `proxy` property is given. The `proxy` property is
+an object where the `key` identifies the target and the `property` is a URI to
+the target.
 
-### .metrics
+### .process(HttpIncoming)
 
-Property that exposes a metric stream.
+Metod for processing a incomming http request. Matches the request against the
```
Comment (digitalsadhu) [inline README.md]: Method .. an incoming HTTP ..

File: README.md [hunk 4]
```diff
@@ -97,36 +110,76 @@ proxy.register({
...
+registered routing targets and proxies the request if a match is found.
+
+Returns a Promise. If the inbound request does match a proxy endpoint the
```
Comment (digitalsadhu) [inline README.md]: ..promise. If the inbound request matches a..

File: README.md [hunk 5]
```diff
@@ -97,36 +110,76 @@ proxy.register({
...
+returned Promise will resolve with `undefined`. If the inbound request does not
```
Comment (digitalsadhu) [inline README.md]: promise

File: README.md [hunk 6]
```diff
@@ -97,36 +110,76 @@ proxy.register({
...
+match a proxy endpoint the returned Promise will resolve with the passed in
```
Comment (digitalsadhu) [inline README.md]: promise

File: README.md [hunk 7]
```diff
@@ -97,36 +110,76 @@ proxy.register({
...
+[HttpIncoming] object.
+
+The method take the following arguments:
```
Comment (digitalsadhu) [inline README.md]: takes

File: README.md [hunk 8]
```diff
@@ -97,36 +110,76 @@ proxy.register({
...
+
+#### HttpIncoming (required)
+
+An instance of a [HttpIncoming] class.
```
Comment (digitalsadhu) [inline README.md]: an

File: README.md [hunk 9]
```diff
@@ -97,36 +110,76 @@ proxy.register({
...
-If any of the entries in the loaded Array are not compatible with the format which `.dump()` exports, they will not be inserted into the cache.
+If any of the entries in the loaded Array are not compatible with the format
+which `.dump()` exports, they will not be inserted into the cache.
 
 Returns an Array with the keys which were inserted into the cache.
 
 ## Where are proxy targets mounted?
 
-To be able to have multible proxy targets in an HTTP server we need to make sure that they do not collide with each other. To prevent so, each proxy target defined is mounted on their own separate namespace in an HTTP server.
+To be able to have multible proxy targets in an HTTP server we need to make sure
```
Comment (digitalsadhu) [inline README.md]: multiple

File: README.md [hunk 10]
```diff
@@ -97,36 +110,76 @@ proxy.register({
...
+that they do not collide with each other. To prevent so, each proxy target
```
Comment (digitalsadhu) [inline README.md]: ... prevent this, ...

File: README.md [hunk 11]
```diff
@@ -97,36 +110,76 @@ proxy.register({
...
+defined is mounted on their own separate namespace in an HTTP server.
```
Comment (digitalsadhu) [inline README.md]: ... on its own separate ...

File: package.json [hunk 12]
```diff
@@ -34,20 +34,18 @@
...
     },
     "devDependencies": {
-        "eslint": "^5.6.1",
+        "eslint": "^5.14.1",
         "eslint-config-airbnb-base": "^13.1.0",
-        "eslint-config-prettier": "^3.1.0",
+        "eslint-config-prettier": "^4.0.0",
         "eslint-plugin-import": "^2.14.0",
         "eslint-plugin-prettier": "^3.0.0",
-        "express": "^4.16.4",
```
Comment (digitalsadhu) [inline package.json]: Sick!!! Proxy without express woot!

File: README.md (hunk 3 above)
Comment (alexanbj) [inline README.md]: incoming

File: package.json (hunk 12 above)
Comment (trygve-lie) [reply package.json]: :fire:

**Pull request merged**